src/View/ViewableData.php

```diff
@@ -556,14 +493,18 @@ protected function objCacheClear()
      * Get the value of a field on this object, automatically inserting the value into any available casting objects
      * that have been specified.
      *
-     * @param string $fieldName
-     * @param array $arguments
-     * @param bool $cache Cache this object
-     * @param string $cacheName a custom cache name
-     * @return object|DBField
-     */
-    public function obj($fieldName, $arguments = [], $cache = false, $cacheName = null)
-    {
+     * @return object|DBField|null The specific object representing the field, or null if there is no
+     * property, method, or dynamic data available for that field.
+     * Note that if there is a property or method that returns null, a relevant DBField instance will
+     * be returned.
+     */
+    public function obj(
+        string $fieldName,
+        array $arguments = [],
+        bool $cache = false,
+        ?string $cacheName = null
+    ): ?object {
```

Comment:
`obj()` can now return `null`.

This makes it incredibly clear the distinction between
- we have a field or method, but it's returning null (will return an object), and
- we have no field or method (will return `null`)

There were a few problems related to removing the iterator where you'd have `<% loop $MissingProperty %>` - and since there's no property it returned a `DBText` by default, and tried to loop over that.
Since we removed the iterator for `ViewableData` that started throwing an error. But it should neither try to iterate over it (there's nothing to iterate over) nor should it throw an error.

Now, because this method explicitly returns `null`, the template engine won't try to iterate over the missing value.